PURL-SPECIFICATION.rst

```diff
@@ -176,25 +178,106 @@ The rules for each component are:
 
 - **qualifiers**:
 
-  - The ``qualifiers`` string is prefixed by a '?' separator when not empty
-  - This '?' is not part of the ``qualifiers``
-  - This is a query string composed of zero or more ``key=value`` pairs each
-    separated by a '&' ampersand. A ``key`` and ``value`` are separated by the equal
-    '=' character
-  - These '&' are not part of the ``key=value`` pairs.
-  - ``key`` must be unique within the keys of the ``qualifiers`` string
-  - ``value`` cannot be an empty string: a ``key=value`` pair with an empty ``value``
-    is the same as no key/value at all for this key
-  - For each pair of ``key`` = ``value``:
-
-    - The ``key`` must be composed only of ASCII letters and numbers, '.', '-' and
-      '_' (period, dash and underscore)
-    - A ``key`` cannot start with a number
-    - A ``key`` must NOT be percent-encoded
-    - A ``key`` is case insensitive. The canonical form is lowercase
-    - A ``key`` cannot contain spaces
-    - A ``value`` must be a percent-encoded string
-    - The '=' separator is neither part of the ``key`` nor of the ``value``
+  - The ``qualifiers`` component MUST be prefixed by a '?' separator when not empty.
+  - The '?' separator is not part of the ``qualifiers`` component.
+  - The ``qualifiers`` component is a query string composed of one or more ``key=value``
+    pairs, each of which is separated by an ampersand '&'. A ``key`` and ``value``
+    are separated by the equal '=' character.
+  - The '&' separator is not part of the ``key`` or the ``value``.
+  - The '=' separator is not part of the ``key`` or the ``value``.
+  - The ``key`` MUST be unique among the keys of the ``qualifiers`` string.
+  - The ``value`` MUST NOT be an empty string: a ``key=value`` pair with an empty ``value``
+    is the same as no ``key=value`` pair at all for this ``key``.
+
+  - For each ``key=value`` pair:
+
+    - ``key``
+
+      - The ``key`` MUST be composed only of ASCII letters and numbers, '.', '-' and
+        '_' (period, dash and underscore).
+      - A ``key`` MUST start with an ASCII letter.
+      - A ``key`` MUST NOT be percent-encoded.
+      - A ``key`` is case insensitive. The canonical form is lowercase.
+
+    - ``value``
+
+      - The ``value`` MUST be composed only of the following characters, encoded
+        as described below and in keeping with RFC 3986 Section 2.2:
```

Comment:
This is confusing and self contradictory. "The `value` MUST be composed only of the following characters", but then the following text defines a set containing all characters.

Reply:
Thanks @matt-phylum -- I don't think it's confusing but in any event the revised update will be simplified and clarified.